packages/editor-ui/src/composables/useRunWorkflow.ts

```diff
@@ -444,6 +443,16 @@ export function useRunWorkflow(useRunWorkflowOpts: { router: ReturnType<typeof u
 
 		await resolveWaitingNodesData();
 
+		// having active execution id causes issue when re-running workflow that has execution data
+		// so we remove it after execution has been resolved
+		const execution = await workflowsStore.getExecution((executionId as string) || '');
+		if (
+			execution &&
+			(execution.finished || ['error', 'canceled', 'crashed', 'success'].includes(execution.status))
```

Comment:
This condition seems to be repeating here and on line 387. Can you please extract it into a `isExecutionResolved` helper function?